subsys/bluetooth/mesh/main.c

```diff
@@ -88,6 +88,28 @@ int bt_mesh_provision(const u8_t net_key[16], u16_t net_idx,
 	return 0;
 }
 
+int bt_mesh_provision_adv(const u8_t uuid[16], u16_t net_idx, u16_t addr,
+			     u8_t attention_duration)
+{
+#if defined(CONFIG_BT_MESH_PROVISIONER) && defined(CONFIG_BT_MESH_PB_ADV)
+	if (!atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+		return -EINVAL;
+	}
+
+	if (bt_mesh_subnet_get(net_idx) == NULL) {
+		return -EINVAL;
+	}
+
+	if (!bt_mesh_is_addr_free(addr)) {
```

Comment:
How will this work if the remote device has multiple elements? Wouldn't it make sense to allocate the address only once you've received the Provisioning Capabilities? In such a case I suppose the `bt_mesh_provision_adv()` wouldn't take an `addr` parameter at all, rather you could report it back (along with number of elements) through the provisioning complete callback.

Reply:
Yea having this check here doesn't really solve the whole problem since we don't yet know the number of elements. Agree that it should be removed.

The `addr` argument could be removed and instead use the first free address that fits. I don't really know if it is of interest to be able to specify an address but maybe it could be an ok solution to pass a `addr_hint` or something. If that is not free then choose the next one that is free.

Reply:
I think there's value in having an address parameter, but perhaps it could be possible to pass invalid address to make it allocate dynamically?

Reply:
Yea I think there is value in being able to specify an address as well. What do you think @jhedberg? Maybe keep the `addr` parameter but if it is 0 (BT_MESH_ADDR_UNASSIGNED) we select the lowest possible address?

Also, in order to have an effective algorithm to find the lowest possible address I think the list of nodes would have to be stored in a sorted list. What do you guys think would be the best solution for finding a free address range?

Reply:
Sounds good regarding keeping the address and the 0 value. Would need to think about the sorting a bit more.

Reply:
I have implemented support for this without having a sorted list but as mentioned in the comment it could be implemented more effectively if they were sorted. I don't think it's that big a deal though.